pkg/leader/leaderelector.go

```diff
@@ -51,22 +52,32 @@ func (f *DummyElector) RunWithLeader(ctx context.Context, run func() error) erro
 var _ ElectorInterface = &Elector{}
 
 type Elector struct {
-	log           logrus.FieldLogger
-	config        Config
-	kube          *kubernetes.Clientset
-	isLeader      bool
-	configMapName string
+	log      logrus.FieldLogger
+	mu       *sync.RWMutex
+	config   Config
+	kube     *kubernetes.Clientset
+	lockName string
+	isLeader bool
 }
 
-func NewElector(kubeClient *kubernetes.Clientset, config Config, configMapName string, logger logrus.FieldLogger) *Elector {
-	logger = logger.WithField("configMap", configMapName)
-	return &Elector{log: logger, config: config, kube: kubeClient, configMapName: configMapName, isLeader: false}
+func NewElector(kubeClient *kubernetes.Clientset, config Config, lockName string, logger logrus.FieldLogger) *Elector {
+	logger = logger.WithField("lock", lockName)
+	mu := &sync.RWMutex{}
+	return &Elector{log: logger, config: config, kube: kubeClient, lockName: lockName, isLeader: false, mu: mu}
 }
 
 func (l *Elector) IsLeader() bool {
+	l.mu.RLock()
```

Comment:
why do you need a mutex? we have only one instance of each thread running locally

Reply:
But the query for IsLeader can come from different threads (like monitor) while the value is set from the elector thread. is it not?

Reply:
Why wasn't it a problem until now?

Reply:
to say the the truth i don't understand why we need this mutex. What's the problem of getting boolean value by multiple threads?